### PR TITLE
Drop many root checks

### PR DESCRIPTION
**Pull request by keszybz**
Split out from #11484.

> The reason the must_be_root() is usually user-friendliness: a short, but explicit message rather than a complicated, non-explicit one...

This is obvious. I explicitly discuss this in the commit message of the first commit. And then in every subsequent commit, why I think the change in that commit makes sense. In general, the changes in this PR make the output more user-friendly, e.g. because `--help` and basic status commands don't request root privileges. If it turns out that the error messages are confusing in some corner cases because of missing privileges, this should be treated as a bug and fixed. (Even for root, it is possible to get permission errors with MAC and namespaces, so we need to have good error reporting anyway.) If any such cases are known, I can add more commits to this PR, and otherwise I can fix them later.

### CODE REVIEW

File: src/udev/udevadm-control.c [hunk 1]
```diff
@@ -73,10 +73,6 @@ int control_main(int argc, char *argv[], void *userdata) {
                 {}
         };
 
-        r = must_be_root();
-        if (r < 0)
-                return r;
```
Comment (yuwata) [inline src/udev/udevadm-control.c:78]: I think the log message shown by `must_be_root()` is simple and useful for users.
So, I prefer to keep that, but in the later stage. Now we parse and simultaneously process based on the parse result. How about to change that first parse, then operate some non-priv operations, and then `must_be_root()`?

**Comment by keszybz** [comment]
> I think the log message shown by must_be_root() is simple and useful for users.
So, I prefer to keep that, but in the later stage.

The problem with the must_be_root() check is that it is overly broad. As described in the individual commit messages, it is often simply false, and not doing the check early allows a lot more functionality to be accessed as non-root. It is better to deliver a more specific (and correct!) message once we get to the point of failure.

Also, even once we get to the point where there was an actual permission issue, with the ongoing separation of root capabilities, MAC checks, namespaces and containerization, simply saying "must be root!" is not precise enough. I'm quite sure that it is better to give a precise error what operation could not be done. It is possible that the user already is root (in some sense), but they still get permission denied.

> Hmm, I do not like to drop MUST_BE_ROOT from verb list.

Is is dropped when it is not used for anything anymore. We generally drop unused functionality like that.

**Comment by yuwata** [comment]
Not sure, but I guess the errors just say `-EPERM` or something. No? The error `-EPERM` is OK, then it suggests the command must run with more privileged account. But, if it returns other errno, then it is hard for users to solve what the problem is...

I agree to show detailed and more suggestive logs. But at least now, I do not think, in general, log messages do not have enough explanation why operations fails. So, just dropping `must_be_root()` is not a good idea.

**Comment by keszybz** [comment]
We expect to get EPERM if an operation fails because of insufficient privs. If we get some unrelated errno that is confusing, this would be a bug on its own. Do you have some examples where our reporting is made worse by the changes here?

**Comment by yuwata** [comment]
OK. Now I am convinced. I have not tested all tools yet. However,
```
$ ./udevadm trigger --action=add /sys/class/net/wlp1s0
```
Does not fail (it exit with 0), and shows nothing. Please fix that.

**Comment by keszybz** [comment]
One more commit for 'udevadm trigger'.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**